yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -510,10 +516,12 @@ private[spark] class Client(
      *   (3) whether to add these resources to the classpath
      */
     val cachedSecondaryJarLinks = ListBuffer.empty[String]
+    val files = sparkConf.get(FILES_TO_DISTRIBUTE).map(p => Utils.resolveURIs(p)).orNull
```

Comment:
There's code in SparkSubmit that does this already, I don't see you removing it, so probably this is not needed? Search for `pathConfigs` in SparkSubmit.scala. It would also be ok if you removed processing of these configs from SparkSubmit.

Also, don't you need to do this for `JARS_TO_DISTRIBUTE` too to be consistent?